nomenclature/codelist.py

```diff
@@ -542,3 +543,28 @@ def from_directory(cls, name: str, path: Path, file_glob_pattern: str = "**/*"):
             mapping[code.name] = code
 
         return cls(name=name, mapping=mapping)
+
+    def hierarchy_filter(self, hierarchy: str) -> "RegionCodeList":
```

Comment:
Suggest to rename that for simplicity (and adding other filter options later).
```suggestion
    def filter(self, hierarchy: str) -> "RegionCodeList":
```